src/main/java/org/lummerland/bamboo/hangoutsNotifications/HangoutsNotificationTransport.java

```diff
@@ -12,52 +15,87 @@
 import com.atlassian.bamboo.notification.NotificationTransport;
 import com.atlassian.bamboo.plan.cache.ImmutablePlan;
 import com.atlassian.bamboo.resultsummary.ResultsSummary;
-import com.google.gson.JsonObject;
+import com.atlassian.bamboo.template.TemplateRenderer;
+import com.google.common.collect.Maps;
 
 public class HangoutsNotificationTransport implements NotificationTransport {
 
 	private static final Logger log = LoggerFactory.getLogger(HangoutsNotificationTransport.class);
 
-	private ImmutablePlan plan;
-	private ResultsSummary resultsSummary;
-	private String webhookUrl;
+	private static final String COLORED_STRING = "<font color='%s'>%s</font>";
+	private static final String FREEMARKER_TEMPLATE = "/templates/hangoutsNotification.ftl";
+
+	private final String webhookUrl;
+	private final ResultsSummary resultsSummary;
+	private final TemplateRenderer templateRenderer;
 
 	private HangoutsNotificationTransport(
 			final String webhookUrl,
-			final ImmutablePlan plan,
-			final ResultsSummary summary) {
+			final ResultsSummary summary,
+			final TemplateRenderer templateRenderer) {
 		this.webhookUrl = webhookUrl;
-		this.plan = plan;
-		this.resultsSummary= summary;
+		this.resultsSummary = summary;
+		this.templateRenderer = templateRenderer;
 	}
 
-	static HangoutsNotificationTransport build(final String webhookUrl, final ImmutablePlan plan, final ResultsSummary summary) {
-		return new HangoutsNotificationTransport(webhookUrl, plan, summary);
+	static HangoutsNotificationTransport build(
+			final String webhookUrl,
+			final ResultsSummary summary,
+			final TemplateRenderer templateRenderer) {
+		return new HangoutsNotificationTransport(webhookUrl, summary, templateRenderer);
 	}
 
 	@Override
 	public void sendNotification(@NotNull final Notification notification) {
-		final JsonObject jsonObject = new JsonObject();
-		jsonObject.addProperty(
-				"text",
-				notification.getIMContent() +
-						"Duration: " + resultsSummary.getDurationDescription());
-
-		CloseableHttpClient client = HttpClients.createDefault();
-		StringEntity requestEntity = new StringEntity(jsonObject.toString(), ContentType.APPLICATION_JSON);
-		HttpPost post = new HttpPost(webhookUrl);
-		post.setEntity(requestEntity);
-		try {
+		try (final CloseableHttpClient client = HttpClients.createDefault()) {
+			final HttpPost post = new HttpPost(webhookUrl);
+			final StringEntity requestEntity = new StringEntity(getMessageJson(), ContentType.APPLICATION_JSON);
+			post.setEntity(requestEntity);
 			log.debug("> send request");
-			CloseableHttpResponse response = client.execute(post);
+			final CloseableHttpResponse response = client.execute(post);
 			log.debug("> got {} response: {}", response.getStatusLine().getStatusCode(), response);
 		}
-		catch (IOException e) {
-			log.error("> Error sending notification: {}", e);
+		catch (final IOException e) {
+			log.error("> Error sending notification", e);
 		}
 	}
+
+	private String getMessageJson() {
```

Comment:
It would be valuable to extract this stuff to split the responsibilities and make things testable. What are the responsibilities here? Collect data to represent a build result could be one. Map this data to some output String could be another. Maybe there is even more :)